lib/functions/main/check.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+###############################################################################
+#
+# string_is_contain()
+# Check for the presence of a word in a string if the field separator is mixed.
+# IFS can be "," comma " " space or ", " both or "\n" the end of a line.
+# usage: if string_is_contain "word1 word2,word3" "word2"; then
+#
+string_is_contain() {
+	[[ -z "$@" ]] || [[ "$#" != 2 ]] && \
+	echo "err: ${FUNCNAME[0]}: Bud argument $@" >&2 && return 1
+	local list=${1// /\\n}
+	echo -e ${list//,/\\n} | grep -q "^$2$"
+}
+
+###############################################################################
+#
+# build_task_is_enabled()
+#
+# $1 - a single task name to check for BUILD_ONLY enablement
+# return:
+#   0 - if BUILD_ONLY if the task name is listed by BUILD_ONLY
+#   1 - otherwise 
+#
+build_task_is_enabled() {
+	string_is_contain "$BUILD_ONLY" "$1"
+}
+
+###############################################################################
+#
+# build_validate_buildOnly()
+#
+# This function validates the list of task names defined by global
+# variable BUILD_ONLY versus the locally defined constant
+# list __all_valid_buildOnly.
+#
+# In case of future extensions, please maintain the list of valid task names
+# only here.
+#
+build_validate_buildOnly() {
+	[[ -z $BUILD_ONLY ]] && display_alert "BUILD_ONLY has empty value" "call dialog" "err" && exit 1
+	local _buildOnly="${BUILD_ONLY}"
+
+	# constant list of all valid BUILD_ONLY task names - can be :comma: or :space: separated
+	local _build_packages="u-boot,kernel,armbian-config,armbian-zsh,plymouth-theme-armbian,armbian-firmware,armbian-bsp"
+	local _build_default="$_build_packages bootstrap"
```

Comment:
Same for `_build_default=$(CONST_build_default)` - this will be needed also in the interactive dialog later.
with:
```
CONST_build_default() {
   echo "$(CONST_build_packages),bootstrap"
}
```

Reply:
@The-going
Stupid question - why don't we start with a constants.sh providing constant values via function names.
All constants used anywhere should be defined in this script.
That would be fail safe for overriding constants during runtime and it would let as grep the code for its usage.

We should define the naming convention for theses functions.

I would vote for:
```
CONST_<main scope/component>_<self explaining variable name>`() {
    echo "constant string"
}
```

Reply:
> why don't we start with a constants.sh providing constant values via function names?

We have two files for this. For what purpose they are placed in different folders remains a mystery to me.
[functions/main/config-prepare.sh](https://github.com/armbian/build/blob/master/lib/functions/main/config-prepare.sh)
[functions/configuration/main-config.sh](https://github.com/armbian/build/blob/master/lib/functions/configuration/main-config.sh)

And I will stick to the naming style of the functions.
```bash
golden_function_name () {
        echo "constant string"
}